deepchecks/utils/performance/partition.py

```diff
@@ -142,11 +154,41 @@ def partition_column(
 
         filters = []
         for i in range(n_large_cats):
-            f = lambda df, val = cat_hist_dict.index[i]: df[column_name] == val
-            filters.append(DeepchecksFilter(f, str(cat_hist_dict.index[i])))
+            f = lambda df, val=cat_hist_dict.index[i]: df[column_name] == val
+            filters.append(DeepchecksFilter([f], str(cat_hist_dict.index[i])))
 
         if len(cat_hist_dict) > n_large_cats:
             f = lambda df, values=cat_hist_dict.index[:n_large_cats]: ~df[column_name].isin(values)
-            filters.append(DeepchecksFilter(f, 'Others'))
+            filters.append(DeepchecksFilter([f], 'Others'))
 
         return filters
+
+
+def convert_tree_leaves_into_filters(tree, feature_names: List[str]) -> List[DeepchecksFilter]:
+    """Extract the leaves from a sklearn tree and covert them into DeepchecksFilters.
+
+    Parameters
+    ----------
+    tree : A sklearn tree
+    feature_names : List[str]
+        The feature names for elements within the tree. Normally it is the column names of the data frame the tree
+           was trained on.
+
+    Returns
+    -------
+    List[DeepchecksFilter]:
+           A list of filters describing the leaves
+    """
+    node_to_feature = [feature_names[i] if i != _tree.TREE_UNDEFINED else None for i in tree.feature]
+
+    def recurse(node_idx, filter_up_to_node):
+        if tree.feature[node_idx] != _tree.TREE_UNDEFINED:
+            left_filter = DeepchecksFilter([lambda df, a=tree.threshold[node_idx]: df[node_to_feature[node_idx]] <= a])
```

Comment:
We use the filter like `dc_filter.filter(df)` where is the `a` parameter coming from?